packages/launchdarkly-react-client-sdk/src/initLDClient.ts

```diff
@@ -9,7 +8,7 @@ interface AllFlagsLDClient {
 
 const initLDClient = async (
   clientSideID: string,
-  user: LDUser = { key: uuid() },
+  user: LDUser = {},
```

Comment:
The problem with doing this is that a user object with no key at all is invalid (unless it has `anonymous:true`, which case you get the "make a UUID or use a cached UUID" behavior). The SDK will refuse to initialize with such a user. Therefore, this would be a breaking change for anyone who omits the user parameter when they call `initLDClient`.

I think we probably shouldn't have allowed the user parameter to be omitted in the first place (it is not optional in the JS SDK or any of the other SDKs), but if we're to be compatible with the existing API, we need to have a valid user object. If you change this to `{ key: '' }`, I believe the SDK will not complain— however, we should test manually to verify this; also, I don't know if events will be counted on the dashboard in that case, so we would need to test that too.

Reply:
Thanks for the insight @eli-darkly.

I may not have time to test this out until next week. If someone else wants to take this over that would be okay with me. Otherwise, I'll comment here when I have time to work on this more.